string file_path
---
string result